network/framework/src/application/interface.rs

```diff
@@ -224,6 +247,38 @@ impl<Message: NetworkMessageTrait> NetworkClientInterface<Message> for NetworkCl
             .send_rpc(peer.peer_id(), rpc_protocol_id, message, rpc_timeout)
             .await?)
     }
+
+    async fn send_to_peer_rpc_raw(
+        &self,
+        message: Bytes,
+        rpc_timeout: Duration,
+        peer: PeerNetworkId,
+    ) -> Result<Message, Error> {
+        let network_sender = self.get_sender_for_network_id(&peer.network_id())?;
+        let rpc_protocol_id =
+            self.get_preferred_protocol_for_peer(&peer, &self.rpc_protocols_and_preferences)?;
+        Ok(network_sender
+            .send_rpc_raw(peer.peer_id(), rpc_protocol_id, message, rpc_timeout)
+            .await?)
+    }
+
+    fn to_bytes<'a>(
+        &self,
+        peers: Vec<PeerNetworkId>,
+        message: Message,
+    ) -> anyhow::Result<HashMap<PeerNetworkId, Box<Bytes>>> {
+        let peers_per_protocol = self.group_peers_by_protocol(peers);
+        // Convert to bytes per protocol
+        let mut bytes_per_peer = HashMap::new();
+        for (protocol_id, peers) in peers_per_protocol {
+            let bytes: Box<Bytes> = Box::new(protocol_id.to_bytes(&message)?.into());
```

Comment:
why it needs to be Box? also it's probably more efficient to serialize per protocol then just clone

Reply:
The Box is so I can share between Peers using the same Protocol and not pay for clone and memory. I only serialize once each protocol.